dask_cuda/device_host_file.py

```diff
@@ -260,6 +260,16 @@ def __delitem__(self, key):
         self.device_keys.discard(key)
         del self.device_buffer[key]
 
+    def evict(self):
+        """Evicts least recently used host buffer (aka, CPU or system memory)
+
+        Implements distributed.spill.ManualEvictProto interface"""
+        try:
+            _, _, weight = self.host_buffer.fast.evict()
+            return weight
+        except Exception:  # We catch all `Exception`s, just like zict.LRU
```

Comment:
I would strongly advise to log it. Otherwise your users will have no clue why they're running out of memory.

Reply:
Currently we can't log appropriately, we may consider this in the future. Could you point me to what Dask currently logs? I'm not sure we even intended to log it this way but could you point to where Dask is logging this currently?

Reply:
https://github.com/dask/distributed/blob/main/distributed/spill.py
In SpillBuffer.evict()
https://github.com/dask/distributed/blob/7a69b5e24c115245005cb83129b4b4b46f05f151/distributed/spill.py#L168-L180

and its context manager SpillBuffer.handle_errors()
https://github.com/dask/distributed/blob/7a69b5e24c115245005cb83129b4b4b46f05f151/distributed/spill.py#L81-L134

Reply:
Thanks @crusaderky for the pointers, for now I opened https://github.com/rapidsai/dask-cuda/issues/873 so we can consider doing the same, we'll also have to think how to add proper logging in Dask-CUDA without just hacking into Distributed's.